src/mgmt/cli_mgmt.c

```diff
@@ -255,11 +255,11 @@ get_attach_info(const char *name, bool all_ranks, struct dc_mgmt_sys_info *info,
 		D_GOTO(out, rc);
 	}
 
-	if (d_agetenv_str(&ofi_interface, "OFI_INTERFACE") == 0)
-		D_INFO("Using client provided OFI_INTERFACE: %s\n", ofi_interface);
+	if (d_agetenv_str(&ofi_interface, "D_INTERFACE") == 0)
+		D_INFO("Using client provided D_INTERFACE: %s\n", ofi_interface);
```

Comment:
What does the behavior end up looking like for a user checking their client logs? I was thinking DAOS should be checking the old variables as well, in order to log correctly here. I see you're logging it in cart, so maybe it's clear enough. Would be nice to see an example.

Reply:
Yeah I was mixed on this, whether to keep both new and old envar readouts in daos or only keep the new one. I ended up removing old usages everywhere, leaving only crt_init() to warn about usage of old envars. 
For your log question are you asking about logs in case when OFI_INTERFACE is set, but not D_INTERFACE?

Reply:
Yes, that's the case I'm thinking of.

Reply:
Ok, was able to get logs and in this scenario OFI_INTERFACE value is ignored. With change in agent above, OFI_INTERFACE is not read out and client sets D_INTERFACE to the value returned from getAttachInfo().

then it calls crt_init() which is now provided both D_INTERFACE and OFI_INTERFACE envariables. In cart init we ignore OFI_INTERFACE if D_INTERFACE is set, and only emit warning about OFI_INTERFACE being set if D_INTERFACE wasnt.

With the above, we end up with no warning/error logs in this scenario. Thoughts on if /what we should change?
 
Log:
`aaoganez@wolf-22 new> unset D_INTERFACE
aaoganez@wolf-22 new> export OFI_INTERFACE=eth1
aaoganez@wolf-22 new>  daos cont create alexpool alexcont --type POSIX
fi   INFO src/gurt/fault_inject.c:777 d_fault_attr_set() Fault Injection attr not set feature not included in build
daos INFO src/client/api/job.c:73 dc_job_init() Using JOBID ENV: DAOS_JOBID
daos INFO src/client/api/job.c:74 dc_job_init() Using JOBID wolf-22.wolf.hpdd.intel.com-1506877
pid:1506877 (daos): numa:0 iface:eth0 dom:eth0 prov:ofi+tcp srx:1
mgmt INFO src/mgmt/cli_mgmt.c:505 dc_mgmt_net_cfg() Setting number of server ranks to 1
mgmt INFO src/mgmt/cli_mgmt.c:531 dc_mgmt_net_cfg() Using server's value for FI_OFI_RXM_USE_SRX: 1
mgmt INFO src/mgmt/cli_mgmt.c:550 dc_mgmt_net_cfg() setenv CRT_TIMEOUT=0
mgmt INFO src/mgmt/cli_mgmt.c:573 dc_mgmt_net_cfg() Network interface: eth0, Domain: eth0
crt  INFO src/cart/crt_init.c:654 crt_init_opt() libcart version 4.9.0 initializing
crt  INFO src/cart/crt_init.c:125 dump_opt() options:
crt  INFO src/cart/crt_init.c:126 dump_opt() crt_timeout = 0
crt  INFO src/cart/crt_init.c:127 dump_opt() max_ctx_num = 1
crt  INFO src/cart/crt_init.c:128 dump_opt() swim_idx = 0
crt  INFO src/cart/crt_init.c:129 dump_opt() provider = (null)
crt  INFO src/cart/crt_init.c:130 dump_opt() interface = (null)
crt  INFO src/cart/crt_init.c:131 dump_opt() domain = (null)
crt  INFO src/cart/crt_init.c:107 dump_envariables() -- ENVARS: --
crt  INFO src/cart/crt_init.c:117 dump_envariables() D_PROVIDER = ofi+tcp
crt  INFO src/cart/crt_init.c:117 dump_envariables() D_INTERFACE = eth0
crt  INFO src/cart/crt_init.c:117 dump_envariables() D_DOMAIN = eth0
crt  INFO src/cart/crt_init.c:117 dump_envariables() D_LOG_FILE = /dev/null
crt  INFO src/cart/crt_init.c:117 dump_envariables() D_LOG_MASK = info
crt  INFO src/cart/crt_init.c:117 dump_envariables() DD_STDERR = debug
crt  INFO src/cart/crt_init.c:117 dump_envariables() CRT_TIMEOUT = 0
crt  INFO src/cart/crt_init.c:117 dump_envariables() OFI_INTERFACE = eth1
crt  INFO src/cart/crt_init.c:117 dump_envariables() CRT_CTX_SHARE_ADDR = 0
crt  INFO src/cart/crt_init.c:117 dump_envariables() FI_OFI_RXM_USE_SRX = 1
crt  INFO src/cart/crt_init.c:117 dump_envariables() CRT_SECONDARY_PROVIDER = 0
`

Reply:
Is this right?
```
crt INFO src/cart/crt_init.c:117 dump_envariables() D_INTERFACE = eth0
...
crt INFO src/cart/crt_init.c:117 dump_envariables() OFI_INTERFACE = eth1
```

Reply:
@frostedcmos I think DAOS needs to check OFI_INTERFACE/etc. and translate accordingly. I have an idea of the changes, I'll push up a PR based on this PR's branch.

Reply:
> Is this right?
> 
> ```
> crt INFO src/cart/crt_init.c:117 dump_envariables() D_INTERFACE = eth0
> ...
> crt INFO src/cart/crt_init.c:117 dump_envariables() OFI_INTERFACE = eth1
> ```

Thats the described behavior of this patch:) looks like Kris has an idea of what daos needs to do

Reply:
#14055 checks both OFI_  and D_ variables in libdaos. libdaos is setting D_INTERFACE for cart's use. @frostedcmos feel free to merge it into this branch if you like the changes. :)